Update BECOMING_A_CORE_CONTRIBUTOR.md (#11337)
PyTorchLightning/pytorch-lightning: .github/CONTRIBUTING.md

We welcome any useful contribution! For your convenience here's a recommended workflow:

> 0. Think about what you want to do - fix a bug, repair docs, etc. If you want to implement a new feature or enhance an existing one.
> 
>    - Start by opening a GitHub issue to explain the feature and the motivation.
>      In the case of features, ask yourself first - Is this NECESSARY for Lightning? There are some PRs which are just
>      purely about adding engineering complexity which have no place in Lightning.
>    - Core contributors will take a look (it might take some time - we are often overloaded with issues!) and discuss it.
>    - Once an agreement was reached - start coding.
> 
> 1. Start your work locally.

Reviewer: ```suggestion
0. Think about what you want to do - fix a bug, repair docs, etc. If you want to implement a new feature or enhance an existing one.

   - Start by opening a GitHub issue to explain the feature and the motivation.
     In the case of features, ask yourself first - Is this NECESSARY for Lightning? There are some PRs that are just
     purely about adding engineering complexity which has no place in Lightning.
   - Core contributors will take a look (it might take some time - we are often overloaded with issues!) and discuss it.
   - Once an agreement was reached - start coding.

1. Start your work locally.
```

   - Create a branch and prepare your changes.
   - Tip: do not work on your master branch directly, it may become complicated when you need to rebase.
   - Tip: give your PR a good name! It will be useful later when you may work on multiple tasks/PRs.

1. Test your code!

   - It is always good practice to start coding by creating a test case, verifying it breaks with current behaviour, and passes with your new changes.
   - Make sure your new tests cover all different edge cases.
   - Make sure all exceptions raised are tested.

1. If your PR is not ready for reviews, but you want to run it on our CI, open a "Draft PR" to let us know you don't need feedback yet.

1. When you feel ready for integrating your work, mark your PR "Ready for review".

   - Your code should be readable and follow the project's design principles.
   - Make sure all tests are passing and any new code is tested for (coverage!).
   - Make sure you link the GitHub issue to your PR.
   - Make sure any docs for that piece of code are updated, or added.
   - The code should be elegant and simple. No over-engineering or hard-to-read code.

>    Do your best but don't sweat about perfection! We do code-review to find any missed items.

Reviewer 2: ```suggestion
   Do your best but don't sweat about perfection! We do code-review to find any missed items. If you need help, don't hesitate to contact the core team.
```

> 1. Use tags in PR name for following cases:

Reviewer: ```suggestion
1. Use tags in PR name for the following cases:
```
